Docs: XmlCollector configuration examples (#8485)
OpenNMS/opennms: docs/modules/reference/pages/performance-data-collection/collectors/xml-examples.adoc

= XmlCollector configuration examples
:description: End-to-end examples of using the XmlCollector to collect time-series metrics from XML and JSON HTTP APIs.

> This page collects worked examples showing how to configure the xref:performance-data-collection/collectors/xml.adoc[XmlCollector] against common HTTP APIs.

Reviewer: words.

```suggestion
This page provides working examples of using the xref:performance-data-collection/collectors/xml.adoc[XmlCollector] against common HTTP APIs.
```

> Each example follows the same shape so you can use one as a template for an integration of your own:

Reviewer: something something Na'vi something Avatar.

```suggestion
Each example follows the same format so you can use one as a template for an integration of your own:
```

Author: LOL

> . *Endpoint(s) and response* -- the URLs being polled and the shape of what they return.

Reviewer: ```suggestion
. *Endpoint(s) and response* -- the URLs being polled and the return format.
```

report.github.stars.columns=ghStars
report.github.stars.type=nodeSnmp
report.github.stars.command=--title="GitHub Stars" \
 --vertical-label="stars" \
 DEF:s={rrd1}:ghStars:AVERAGE \
 LINE2:s#1f78b4:"Stars" \
 GPRINT:s:LAST:"Cur\\: %8.0lf"

report.github.forks.name=GitHub Forks
report.github.forks.columns=ghForks
report.github.forks.type=nodeSnmp
report.github.forks.command=--title="GitHub Forks" \
 --vertical-label="forks" \
 DEF:f={rrd1}:ghForks:AVERAGE \
 LINE2:f#33a02c:"Forks" \
 GPRINT:f:LAST:"Cur\\: %8.0lf"

report.github.watchers.name=GitHub Watchers
report.github.watchers.columns=ghWatchers
report.github.watchers.type=nodeSnmp
report.github.watchers.command=--title="GitHub Watchers" \
 --vertical-label="watchers" \
 DEF:w={rrd1}:ghWatchers:AVERAGE \
 LINE2:w#6a3d9a:"Watchers" \
 GPRINT:w:LAST:"Cur\\: %8.0lf"

report.github.openIssues.name=GitHub Open Issues + PRs
report.github.openIssues.columns=ghOpenIssues
report.github.openIssues.type=nodeSnmp
report.github.openIssues.command=--title="GitHub Open Issues + PRs" \
 --vertical-label="count" \
 DEF:i={rrd1}:ghOpenIssues:AVERAGE \
 LINE2:i#e31a1c:"Open" \
 GPRINT:i:LAST:"Cur\\: %5.0lf"
----

=== Verification

> After reloading collectd (`send-event.pl uei.opennms.org/internal/reloadDaemonConfig --parm 'daemonName Collectd'`), test a one-shot collection from the xref:reference:karaf-shell/karaf-shell.adoc[Karaf shell] without waiting for the next scheduled cycle:

Reviewer: ```suggestion
After reloading collectd (`send-event.pl uei.opennms.org/internal/reloadDaemonConfig --parm 'daemonName Collectd'`), test an xref:deep-dive/performance-data-collection/shell/adhoc-collection.adoc[Ad-Hoc collection] from the xref:reference:karaf-shell/karaf-shell.adoc[Karaf shell] without waiting for the next scheduled cycle:
```
